### ISSUE TITLE
Doc system emits malformed ReST

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/julia#12746]
Lately the travis log is too long to be displayed normally, which is kind of annoying and unnecessary. Most of the new output seems to be doc-related warnings.

**Comment by jiahao** [comment]
The doc system is emitting malformed ReST. There are many missing newlines which are needed to demarcate the beginning and end of a ReST directive block. For example

```
.. code-block:: julia
    toprev(dt::TimeType,dow::Int;same::Bool=false) -> TimeType
```

is parsed as a runon list of 4 arguments `julia`, `toprev...`, `->` and `TimeType` to the `code-block` directive. There needs to be a newline in between or else Sphinx throws an error `ERROR: Error in "code-block" directive: maximum 1 argument(s) allowed, 4 supplied.`.

Similarly

```
.. code-block:: julia

    toprev(dt::TimeType,dow::Int;same::Bool=false) -> TimeType
blah blah blah
```

is an error; there needs to be a newline or else Sphinx complains `WARNING: Explicit markup ends without a blank line; unexpected unindent.`.

I've mentioned this to @one-more-minute on previous occasions; hopefully it is being worked on.

**Comment by JeffBezanson** [comment]
Much better now, but there are still tons of ReST warnings in the build log.

**Comment by jakebolewski** [comment]
closed by #12835

**Comment by jiahao** [comment]
:fireworks:
